File: Pinball App Android/app/src/main/java/com/pillyliu/pinprofandroid/stats/StatsModels.kt
package com.pillyliu.pinprofandroid.stats

internal data class ScoreRow(
    val id: Int,
    val season: String,
    val bankNumber: Int,
    val player: String,
    val machine: String,
    val rawScore: Double,
    val points: Double,
)

internal data class StatPlayerLabel(
    val rawPlayer: String,
    val season: String?,
)

internal data class StatResult(
    val count: Int,
    val low: Double?,
    val lowPlayer: StatPlayerLabel?,
    val high: Double?,
    val highPlayer: StatPlayerLabel?,
    val mean: Double?,
    val median: Double?,
    val std: Double?,
)

internal data class StatsTableWidths(
    val season: Int,
    val bank: Int,
    val player: Int,
    val machine: Int,
    val score: Int,
    val points: Int,
)

internal data class StatsLoadResult(
    val rows: List<ScoreRow>,
    val updatedAtMs: Long?,
)

File: Pinball App 2/Pinball App 2/stats/StatsModels.swift
import Foundation

struct ScoreRow: Identifiable {
    let id: Int
    let season: String
    let bankNumber: Int
    let player: String
    let machine: String
    let rawScore: Double
    let points: Double
}

struct StatPlayerLabel {
    let rawPlayer: String
    let season: String?
}

struct StatResult {
    let count: Int
    let low: Double?
    let lowPlayer: StatPlayerLabel?
    let high: Double?
    let highPlayer: StatPlayerLabel?
    let mean: Double?
    let median: Double?
    let std: Double?

    static let empty = StatResult(
        count: 0,
        low: nil,
        lowPlayer: nil,
        high: nil,
        highPlayer: nil,
        mean: nil,
        median: nil,
        std: nil
    )
}
